airbyte-cdk/python/airbyte_cdk/sources/configurable/retrievers/simple_retriever.py

```diff
@@ -0,0 +1,244 @@
+        Any option listed in https://docs.python-requests.org/en/latest/api/#requests.adapters.BaseAdapter.send for can be returned from
+        this method. Note that these options do not conflict with request-level options such as headers, request params, etc..
+        """
+        return self._requester.request_kwargs(stream_state, stream_slice, next_page_token)
+
+    def path(
+        self, *, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return self._requester.get_path(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        """
+        Specifies the query parameters that should be set on an outgoing HTTP request given the inputs.
+
+        E.g: you might want to define query parameters for paging if next_page_token is not None.
+        """
+        return self._requester.request_params(stream_state, stream_slice, next_page_token)
+
+    @property
+    def cache_filename(self):
+        """
+        Return the name of cache file
+        """
+        return self._requester.cache_filename
+
+    @property
+    def use_cache(self):
+        """
+        If True, all records will be cached.
+        """
+        return self._requester.use_cache
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        *,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        self._last_response = response
+        records = self._extractor.extract_records(response)
+        self._last_records = records
+        return records
+
+    @property
+    def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
+        return self._primary_key
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        Specifies a pagination strategy.
+
+        The value returned from this method is passed to most other methods in this class. Use it to form a request e.g: set headers or query params.
+
+        :return: The token for the next page from the input response object. Returning None means there are no more pages to read in this response.
+        """
+        return self._paginator.next_page_token(response, self._last_records)
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        records = [r for r in HttpStream.read_records(self, sync_mode, cursor_field, stream_slice, stream_state)]
```

Comment:
does this hold all records in memory? could we do it in an online way instead?

Reply:
yes, done!